contracts/rbn-staking/FeeCustody.sol

```diff
@@ -0,0 +1,425 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.11;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import {SafeMath} from "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "../interfaces/IFeeDistributor.sol";
+import "../interfaces/IChainlink.sol";
+import "../interfaces/IWETH.sol";
+
+import "@uniswap/v3-periphery/contracts/interfaces/ISwapRouter.sol";
+import "@uniswap/v3-periphery/contracts/libraries/TransferHelper.sol";
+
+/** @title FeeCustody
+    @notice Custody Contract for Ribbon Vault Management / Performance Fees
+ */
+
+contract FeeCustody is Ownable {
+  using SafeERC20 for IERC20;
+  using SafeMath for uint256;
+
+  address public constant WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+  // WETH Distribution Token
+  IERC20 public distributionToken = IERC20(WETH);
+  // Protocol revenue recipient
+  address public protocolRevenueRecipient;
+  // Address of fee distributor contract for RBN lockers to claim
+  IFeeDistributor public feeDistributor;
+
+  // % allocation (0 - 100%) from protocol revenue to allocate to RBN lockers.
+  // 2 decimals. ex: 10% = 1000
+  uint256 public pctAllocationForRBNLockers;
+
+  uint256 public constant TOTAL_PCT = 10000; // Equals 100%
+  ISwapRouter public constant UNIV3_SWAP_ROUTER =
+    ISwapRouter(0x68b3465833fb72A70ecDF485E0e4C7bD8665Fc45);
+
+  // Intermediary path asset for univ3 swaps.
+  // Empty if direct pool swap between asset and distribution asset
+  mapping(address => bytes) public intermediaryPath;
+
+  // Oracle between asset/usd pair for total
+  // reward approximation across all assets earned
+  mapping(address => address) public oracles;
+
+  address[1000] public assets;
+  // Index of empty slot in assets array
+  uint256 public lastAssetIdx;
```

Comment:
Since we're also tracking the length of `assets` in a separate variable, what's the rationale for not just using dynamic length arrays `address[] public assets`?

Doing `assets.length` is cheaper than setting/getting `lastAssetIdx`

Reply:
changed